src/ucai/core/envs/databricks_env_vars.py

```diff
@@ -0,0 +1,55 @@
+import os
+from typing import Optional
+
+
+class _EnvironmentVariable:
+    def __init__(self, name: str, default_value: str, description: Optional[str] = None):
+        self.name = name
+        self.default_value = str(default_value)
+        self.description = description
+
+    def get(self) -> str:
+        return os.getenv(self.name, self.default_value)
+
+    def set(self, value: str) -> None:
+        os.environ[self.name] = str(value)
+
+    def remove(self) -> None:
+        os.environ.pop(self.name, None)
+
+    def __repr__(self) -> str:
+        return f"Environment variable for {self.name}. Default value: {self.default_value}. " + (
+            f"Usage: {self.description}" if self.description else ""
+        )
+
+
+UCAI_DATABRICKS_WAREHOUSE_EXECUTE_FUNCTION_WAIT_TIMEOUT = _EnvironmentVariable(
+    "UCAI_DATABRICKS_WAREHOUSE_EXECUTE_FUNCTION_WAIT_TIMEOUT",
+    "30s",
+    "The time in seconds the call will wait for the function to execute using Databricks client with warehouse_id; "
+    "the value should be set as `Ns`, where `N` can be set to 0 or to a value between 5 and 50.",
+)
+UCAI_DATABRICKS_WAREHOUSE_EXECUTE_FUNCTION_ROW_LIMIT = _EnvironmentVariable(
+    "UCAI_DATABRICKS_WAREHOUSE_EXECUTE_FUNCTION_ROW_LIMIT",
+    "100",
+    "Maximum number of rows of the function execution result using Databricks client with warehouse_id; "
+    "it also sets the `truncated` field in the response to indicate whether the result was trimmed due to the limit or not.",
+)
+UCAI_DATABRICKS_WAREHOUSE_EXECUTE_FUNCTION_BYTE_LIMIT = _EnvironmentVariable(
+    "UCAI_DATABRICKS_WAREHOUSE_EXECUTE_FUNCTION_BYTE_LIMIT",
+    "4096",
+    "Maximum byte size of the function execution result using Databricks client with warehouse_id; "
+    "If the result was truncated due to the byte limit, then `truncated` in the response is set to `true`.",
+)
+UCAI_DATABRICKS_WAREHOUSE_RETRY_TIMEOUT = _EnvironmentVariable(
```

Comment:
Instead of handling two temporal settings for these timeouts, would it be a little less confusing to have a retry count that iterates on `UCAI_DATABRICKS_WAREHOUSE_EXECUTE_FUNCTION_WAIT_TIMEOUT`? 

I'm curious what the behavior would be if I set that wait timeout value to say 45s, the RETRY timeout to 60... would I effectively abort on the first retry if the total execution time of my function is 20s, even if it was going to return, or do I only abort a retry if the total elapsed time exceeds this value? 
Would it be more intuitive to just say "I want to retry at max 3 times where if I'm waiting for more than 20s on each one, I want to retry with some delay between the retries"?

Reply:
If the statement execution completes within `WAIT_TIMEOUT` then the response state should be SUCCEEDED, and it won't enter the retry loop; We only retry if the statement is still executing after `WAIT_TIMEOUT`, and try to fetch the response until `RETRY_TIMEOUT` is reached.
Sometimes people could set WAIT_TIMEOUT to a large number as it's the total execution time, but we might not want to wait the same time for retrieving response? Exponential retry time seems make more sense 🤔 
If this is confusing, I can add documentation in the README to explain them!